From 4df7e141f6a30f26bbe27ebe73def56e8e8ce715 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Mon, 19 Sep 2022 15:51:10 +0300
Subject: [PATCH 31/62] xen_arm: Init grants region as well

TODO: As this action is needed for HVM guests maybe it would
be better to move that to some common function called for both
Arm and x86 (xen_register_ioreq())?

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 hw/arm/xen_arm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/arm/xen_arm.c b/hw/arm/xen_arm.c
index dc1c57adf5..a193f74e4c 100644
--- a/hw/arm/xen_arm.c
+++ b/hw/arm/xen_arm.c
@@ -139,6 +139,9 @@ static void xen_init_ram(MachineState *machine)
         DPRINTF("Initialized region xen.ram.hi: base 0x%llx size 0x%lx\n",
                 GUEST_RAM1_BASE, ram_size[1]);
     }
+
+    /* Add grant mappings as a pseudo RAM region. */
+    ram_grants = xen_init_grant_ram();
 }
 
 static void xen_create_pcie(XenArmState *xam)
-- 
2.34.1

